api/src/test/java/org/accula/api/routers/GreetingRouterTest.java

```diff
@@ -1,39 +1,42 @@
 package org.accula.api.routers;
 
-import org.junit.Test;
-import org.junit.runner.RunWith;
+import org.accula.api.handlers.GreetingHandler;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest;
+import org.springframework.context.annotation.Import;
 import org.springframework.test.web.reactive.server.WebTestClient;
 import org.springframework.web.reactive.function.server.RouterFunction;
 import org.springframework.web.reactive.function.server.ServerResponse;
 
-@SpringBootTest
-@RunWith(SpringRunner.class)
+@WebFluxTest
+@Import({GreetingRouter.class, GreetingHandler.class})
 public final class GreetingRouterTest {
     private static final String GREETING = "ACCULA is greeting you, Alice";
     private static final String ERROR = "Missing required query param \"name\"";
 
     @Autowired
     private RouterFunction<ServerResponse> greetingRoute;
+    private WebTestClient client;
+
+    @BeforeEach
+    public void setUp() {
+        client = WebTestClient
+                .bindToRouterFunction(greetingRoute)
+                .build();
+    }
 
     @Test
     public void testGreetingRouteOk() {
-        WebTestClient
-                .bindToRouterFunction(greetingRoute)
-                .build()
-                .get().uri("/greet?name=Alice").exchange()
+        client.get().uri("/greet?name=Alice").exchange()
                 .expectStatus().isOk()
                 .expectBody(String.class).isEqualTo(GREETING);
     }
 
     @Test
     public void testGreetingRouteBadRequest() {
-        WebTestClient
-                .bindToRouterFunction(greetingRoute)
-                .build()
-                .get().uri("/greet?name=").exchange()
+        client.get().uri("/greet?name=").exchange()
```

Comment:
Put `exchange()` on a new line, to follow the Spring [style](https://docs.spring.io/spring/docs/current/spring-framework-reference/pdf/testing-webtestclient.pdf)

Reply:
Thanks! Did it